### PR TITLE
Feature: Settings sidebar revamp

### LINKED ISSUE #2169: Settings: Sidebar Revamp
# Intro
We will be updating the sidebar in the Settings app to match the new styling we have introduced in Content, Media, Schema, Reports, and Apps. 

image

In addition to this we will be introducing the ability to filter the sidebar for settings. 

image

# Figma
[See Figma File](https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/Zesty-3.0-Revamp?type=design&node-id=2159%3A157307&mode=design&t=mJwpH0Qa8gCrV3iW-1)

### PR DESCRIPTION
**Pull request by theofficialnar**
Closes #2169 

### Preview
settings

### CODE REVIEW

File: src/shell/services/types.ts [hunk 1]
```diff
@@ -464,3 +464,14 @@ export interface InstalledApp {
   updatedAt: string;
   url: string;
 }
+
+export interface StyleCategory {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/shell/services/types.ts]: Updated

File: src/apps/settings/src/app/components/Nav/SettingsNav.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        />
+      </Box>
+    </AppSideBar>
+  );
+};
+
+interface HeaderComponentProps {
+  title: string;
+}
+const HeaderComponent: FC<HeaderComponentProps> = ({ title }) => {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/settings/src/app/components/Nav/SettingsNav.tsx]: Updated

File: src/apps/settings/src/app/components/Nav/SettingsNav.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+              : FONTS_CAT
+          }
+          selected={location.pathname}
+        />
+      </Box>
+    </AppSideBar>
+  );
+};
+
+interface HeaderComponentProps {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/settings/src/app/components/Nav/SettingsNav.tsx]: Updated

**Review by zcolah — CHANGES_REQUESTED** [review]
Great job as always @theofficialnar. Two key things to fix. 

1. Empty State defined is not present 
CleanShot 2023-09-18 at 13 54 24@2x

2. Please make the default width of all sidebars to be the minimum width rather than being a width between the min and max.

**Pull request merged**